src/CoreWCF.WebHttp/src/CoreWCF/Configuration/ServiceModelWebServiceBuilderExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using CoreWCF.Description;
+using CoreWCF.OpenApi.Middleware;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CoreWCF.Configuration
+{
+    public static class ServiceModelWebServiceBuilderExtensions
+    {
+        public static IServiceBuilder AddServiceWebEndpoint<TService, TContract>(
+            this IServiceBuilder builder,
+            IApplicationBuilder app,
+            WebHttpBinding binding,
```

Comment:
You should be able to get rid of the IApplicationBuilder being passed in. You really only need the IServiceProvider. As we discussed, add an overload to AddServiceEndpoint to have a delegate which takes the ServiceEndpoint. You will be adding an IServiceBehavior to DI which will add WebHttpBehavior to all appropriate endpoints. As this IServiceBehavior will be injected, you can capture the IServiceProvider by having it as a parameter to the constructor of the service behavior. You can then propagate that to the WebHttpBehavior. In your delegate that you pass in these methods, you will be able to retrieve the WebHttpBehavior and then have access to the IServiceProvider.

Reply:
This has been addressed.